### PR TITLE
Fix compiling with older Intel compilers

### PR DESCRIPTION
**Pull request by masterleinad**
This PR fixes compiling with Intel 2016 and 2017 for me. The problematic PR was #5090, more specifically commit dc2d367d062ea093e036172e003ad90d4cb030ee. In particular, the second part
```
std::is_base_of<FiniteElement<dim, spacedim>, typename std::decay<FEPairs>::type>::value)
```
in
```
template <class... FEPairs,
            typename = typename enable_if_all<
              (std::is_same<typename std::decay<FEPairs>::type,
                            std::pair<std::unique_ptr<FiniteElement<dim, spacedim>>, unsigned int>>::value
               ||
               std::is_base_of<FiniteElement<dim, spacedim>,
                               typename std::decay<FEPairs>::type>::value)
              ...
              >::type
            >
  FESystem (FEPairs &&... fe_pairs);
```
caused internal compiler errors in the older compilers even if we were not using them at all. (In fact, it dependends on the existence of other member variables for these problems to be visable). 
Since we know that these constructors are not working properly for anything before `Intel 19`, I decided to just disable them all together and not just the problematic part.
I didn't touch the tests respective tests `fe_system_03` and `fe_system_04` since the behavior doesn't change anyway.

I tested compiling with `ICC-16` and `ICC-17` and additionally ran the `FESystem` tests for `ICC 19`.

### CODE REVIEW

File: include/deal.II/fe/fe_system.h [hunk 1]
```diff
@@ -475,7 +475,11 @@ class FESystem : public FiniteElement<dim, spacedim>
    * In other words, if no multiplicity for an element is explicitly specified
    * via the exponentiation operation, then it is assumed to be one (as one
    * would have expected).
+   *
+   * @warning This feature is not available for Intel compilers
+   * prior to version 19.0
    */
+#  if !defined(__INTEL_COMPILER) || __INTEL_COMPILER >= 1900
```
Comment (davydden) [inline include/deal.II/fe/fe_system.h]: I don’t have issues with 18.02

Comment (davydden) [reply include/deal.II/fe/fe_system.h]: Please lower the requirement to 18

Comment (masterleinad) [reply include/deal.II/fe/fe_system.h]: So you can run `fe/system_03` and `fe/system_04` successfully?

Comment (davydden) [reply include/deal.II/fe/fe_system.h]: I did not run any tests (that’s a local cluster, admins wouldn’t appreciate it I think), but compilation is ok.

Comment (masterleinad) [reply include/deal.II/fe/fe_system.h]: Yes, of course, I was testing with ICC-18 regularly, but these tests always failed. Hence, I want to document that people can't expect these constructors to work before ICC-19 (for which the tests are indeed passing). Having a different restriction in the source code seems to be inconsistent and the result is the same. The compiler never considered these constructors in overload resolution properly.

File: include/deal.II/fe/fe_system.h [hunk 2]
```diff
@@ -475,7 +475,11 @@ class FESystem : public FiniteElement<dim, spacedim>
    * In other words, if no multiplicity for an element is explicitly specified
    * via the exponentiation operation, then it is assumed to be one (as one
    * would have expected).
+   *
+   * @warning This feature is not available for Intel compilers
+   * prior to version 19.0
```
Comment (bangerth) [inline include/deal.II/fe/fe_system.h]: Maybe say why?

Also, I think you want to put the `#if` *above* the comment

File: include/deal.II/fe/fe_system.h (hunk 1 above)
Comment (davydden) [reply include/deal.II/fe/fe_system.h]: Thanks for explanation

**Comment by masterleinad** [comment]
> I guess you weren't able to find a small testcase from which we can create a cmake check?

No, not really.

**Pull request merged**